packages/commonwealth/server/workers/evmChainEvents/logProcessing.ts

```diff
@@ -64,11 +66,10 @@ export async function getLogs({
     return { logs: [], lastBlockNum: endBlock };
   }
 
-  if (endBlock - startBlock > 500) {
-    // limit the number of blocks to fetch to 500 to avoid rate limiting on some EVM nodes like Celo
-    // this should eventually be configured on the ChainNodes table by rpc since each rpc has different
-    // rate limits e.g. Alchemy has a limit of 10k logs while Celo public nodes have a limit of 500.
-    startBlock = endBlock - 500;
+  // limit the number of blocks to fetch to avoid rate limiting on some public EVM nodes like Celo
+  // maxBlockRange = -1 indicates there is no block range limit
+  if (maxBlockRange !== -1 && endBlock - startBlock > maxBlockRange) {
+    startBlock = endBlock - maxBlockRange;
```

Comment:
One thing I find a bit odd is that it's updating the *start* block based on the range. If we're processing a new chunk of blocks, then shouldn't the start remain the same and the *end* be adjusted based on range?

Reply:
cc @timolegros can you explain?

Reply:
The one issue I could envision is that if a chain node is rate limiting our requests, updating the end block instead of the start block would mean that we could be consistently behind the latest block and therefore emitting old notifications rather than missing some notifications and emitting recent notifications. At the time when chain-events was created notifications were the primary purpose but now that contests have been introduced, never missing an event is no longer acceptable. Since we don't listen to rate-limited chain nodes anymore this is no longer a concern since this PR introduces infinite block ranges for chain nodes that aren't rate-limited so the change is also not necessary in that regard.

Additionally, since this PR has been tested and we will be removing the chain-events service very soon we should merge as is instead of updating this and having to re-test and deal with updating the chain-event integration tests.